quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/PropertyMappingInterceptor.java

```diff
@@ -82,29 +81,6 @@ public ConfigValue getValue(ConfigSourceInterceptorContext context, String name)
         if (Boolean.TRUE.equals(disable.get())) {
             return context.proceed(name);
         }
-        ConfigValue value = PropertyMappers.getValue(context, name);
-
-        if (value == null || value.getValue() == null) {
-            return null;
-        }
-
-        if (!value.getValue().contains("${")) {
-            return value;
-        }
-
-        // Our mappers might have returned a value containing an expression ${...}.
-        // However, ExpressionConfigSourceInterceptor was already executed before (to expand e.g. env vars in config file).
-        // Hence, we need to manually resolve these expressions here. Not ideal, but there's no other way (at least I haven't found one).
-        return value.withValue(
-                StringPropertyReplacer.replaceProperties(value.getValue(),
-                        property -> {
-                            ConfigValue prop = context.proceed(property);
-
-                            if (prop == null) {
-                                return null;
-                            }
-
-                            return prop.getValue();
-                        }));
+        return PropertyMappers.getValue(context, name);
```

Comment:
It's a little complicated with the various paths. In general we can treat ConfigValue==null and ConfigValue with a null as the same, it's just up to the smallrye logic to honor their stated contract - some methods guarantee returning a ConfigValue, so it will just wrap nulls.